client/core/trade.go

```diff
@@ -288,15 +287,12 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 				},
 			},
 		}
-		err := t.db.UpdateMatch(&match.MetaMatch)
-		if err != nil {
-			return err
-		}
 		// First check that this isn't a match on its own cancel order. I'm not crazy
 		// about this, but I am detecting this case right now based on the Address
 		// field being an empty string.
 		isCancel := t.cancel != nil && msgMatch.Address == ""
 		if isCancel {
+			match.setStatus(order.MatchComplete)
```

Comment:
Was thinking about this while working on https://github.com/decred/dcrdex/pull/319.  Glad my client/core thinking was right here.